### PR TITLE
Script/Creature: script Terokk's encounter

### LINKED ISSUE #2029: DB/Quest: Terokk's Downfall
http://www.wowhead.com/quest=11073/terokks-downfall 

the quest should allow to summon the boss through the bones at the scene ... just that, talking to the bones, does not seem able to summon the boss

http://www.youtube.com/watch?v=71dF4s9ZCRM
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/256949-quest-terokk-s-downfall?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Rothend**
**Changes proposed:**

This PR scripts [Terokk](https://www.wowhead.com/npc=21838) (summoned boss in Skettis, Terokkar Forest).

The script itself is all SAI, with a spell script in core to make [Skyguard Flare](https://www.wowhead.com/spell=40655) target the ground when picking a random dest.

Waypoints for the [Skyguard Aces](https://www.wowhead.com/npc=23377) are guessed based on Retail videos.

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [x] master

**Issues addressed:** Closes #2029

**Tests performed:** tested, works.

### CODE REVIEW

**Comment by offl** [comment]
Maybe https://github.com/cmangos/mangos-tbc/blob/ce04ac7170a1eb6974d27b190fa171787bd6bca2/src/game/AI/ScriptDevAI/scripts/outland/boss_terokk.cpp contains proper waypoints?

**Comment by Aokromes** [comment]
tip: to make circle happy put guid 9999999 xd

File: sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=22 AND `SourceEntry`=21838;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
+(13, 3, 40722, 0, 0, 1, 0, 40726, 0, 0, 1, 0, 0, "", "Will of the Arakkoa God does not affect targets with Chosen One aura"),
+(22, 13, 21838, 0, 0, 1, 1, 40657, 1, 0, 1, 0, 0, "", "SAI executes only if Terokk does not have Ancient Flames aura AND"),
+(22, 13, 21838, 0, 0, 1, 1, 40733, 0, 0, 1, 0, 0, "", "SAI executes only if Terokk does not have Divine Shield aura");
+
+-- Will of the Arakkoa God is negative
+DELETE FROM `spell_custom_attr` WHERE `entry`=40722;
+INSERT INTO `spell_custom_attr` (`entry`, `attributes`) VALUES
+(40722, 4096+8192);
```
Comment (Killyana) [inline sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql:96]: It's better to fix this in core side, we have lots of similar issues

Comment (Rothend) [reply sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql]: Done! I checked the dbc and the only thing that could tell us it's a negative spell is the target type (TARGET_UNIT_SRC_AREA_ENEMY), maybe that can be useful for a more general fix.

**Comment by Rothend** [comment]
Done!

@offl I'm not familiar with (C)MaNGOS, tried to get the waypoints from there but can't find anything. Checked script_waypoint after applying full db+updates+ACID. Any tip? :P

File: sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql (hunk 1 above)
Comment (Killyana) [reply sql/updates/world/3.3.5/9999_99_99_99_world_terokk.sql]: @Jildor does this requires the same fix as https://github.com/TrinityCore/TrinityCore/pull/24862/files ?

**Comment by Killyana** [comment]
Cmangos use : `creature_movement_template` and `creature_movement` for waypoints

**Comment by illfated** [comment]
BTW, seeing that

**Comment by illfated** [comment]
BTW, does the error in **ci/circleci: nopch** indicate that the .cpp file needs the addition of the following include?
```cpp
#include "Map.h"
```
… or is anything else missing in the script?

**Comment by Rothend** [comment]
Yeah, forgot to check for nopch! Done.

**Comment by illfated** [comment]
Sounds good to me. Also, thank you for your quite impressive amount of contributions during this short time. Nice work.

**Comment by jackpoz** [comment]
is this PR ready to be merged (other than the 3 guid slot) ?

**Comment by illfated** [comment]
Cosmetic detail/suggestion, or for the future: using old-school naming conventions, this PR could be titled something like e.g. 
- **Scripts/Terokkar Forest: Terokk's Downfall** 
- **Creature/Script: summoning Terokk**

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**